src/lib/components/controls/ControlsList.svelte

```diff
@@ -119,15 +119,83 @@
 				results = results.filter((control) => {
 					// Control must match all filters
 					return $activeFilters.every(filter => {
+						// Handle mapping-related fields specially
+						if (filter.fieldName === 'has_mappings') {
+							const hasMappings = control.mappings && control.mappings.length > 0;
+							const filterValue = getFilterValue(filter.value);
+							switch (filter.operator) {
+								case 'equals':
+									return hasMappings === (filterValue === 'true' || filterValue === true);
+								case 'not_equals':
+									return hasMappings !== (filterValue === 'true' || filterValue === true);
+								case 'exists':
+									return hasMappings;
+								case 'not_exists':
+									return !hasMappings;
+								default:
+									return true;
+							}
+						}
+
+						if (filter.fieldName === 'mapping_count') {
+							const mappingCount = control.mappings ? control.mappings.length : 0;
+							const filterValue = getFilterValue(filter.value);
+							switch (filter.operator) {
+								case 'equals':
+									return mappingCount === Number(filterValue);
+								case 'not_equals':
+									return mappingCount !== Number(filterValue);
+								case 'exists':
+									return mappingCount > 0;
+								case 'not_exists':
+									return mappingCount === 0;
+								default:
+									return true;
+							}
+						}
+
+						if (filter.fieldName === 'mapping_status') {
+							if (!control.mappings || control.mappings.length === 0) {
+								return filter.operator === 'not_exists';
+							}
+
+							const filterValue = getFilterValue(filter.value);
+							const hasStatus = control.mappings.some(mapping => {
+								switch (filter.operator) {
+									case 'equals':
+										return mapping.status === filterValue;
+									case 'not_equals':
+										return mapping.status !== filterValue;
+									case 'includes':
+										return (typeof mapping.status === 'string' ? mapping.status.toLowerCase() : '').includes(String(filterValue).toLowerCase());
+									case 'not_includes':
+										return !(typeof mapping.status === 'string' ? mapping.status.toLowerCase() : '').includes(String(filterValue).toLowerCase());
+									default:
+										return true;
+								}
+							});
```

Comment:
Since the UI "filterValue" is a select box, there's not really any clear value from the contains/not contains right now in my opinion?

I could see an approach where "equals" is more of "every control has this status" and "contains" is "one control has this status".

But if we want to start simpler maybe we just allow for equals/not equals only and "some mapping" satisfies it?

Reply:
i agree, forced just equals in the operator